integration_test/cases/repo.exs

```diff
@@ -1075,25 +1075,25 @@ defmodule Ecto.Integration.RepoTest do
     test "Repo.insert_all upserts and fills in placeholders with conditioned on_conflict query" do
       do_not_update_title = "don't touch me"
 
-      visits_value =
-        from p in Post, where: p.public == ^true and p.id > ^0, select: p.visits, limit: 1
+      posted_value =
+        from p in Post, where: p.public == ^true and p.id > ^0, select: p.posted, limit: 1
 
       on_conflict =
         from p in Post, update: [set: [title: "updated"]], where: p.title != ^do_not_update_title
 
-      placeholders = %{posted: Date.utc_today(), title: "title"}
+      placeholders = %{visits: 1, title: "title"}
 
       post1 = [
-        visits: visits_value,
+        visits: {:placeholder, :visits},
         title: {:placeholder, :title},
         uuid: Ecto.UUID.generate(),
-        posted: {:placeholder, :posted}
+        posted: posted_value
```

Comment:
This test should have caught the bug, but previously the placeholder values were being counted before the `visits` query in `plan_query_in_rows`. So the effect was the same as if the placeholder count was passed into this function.

When I make it so that the query comes before the placeholder, the test fails.